### PR TITLE
feat: Request and use data from `fastify-overview` endpoint

### LINKED ISSUE #5: Request and use data from fastify-overview endpoint

### PR DESCRIPTION
**Pull request by raul-saez**
Closes #5 

## What this PR contains

- all the usual setup with eslint, commitlint, dependabot and so on
- the main feature of actually bundling this into a fastify plugin which grabs the data from fastify-overview and renders the UI

## What it doesn't contain

No changes about how the actual UI works, I didn't touch the client code apart from formatting and the grabbing the data from the route instead of hardcoded

## How this works

The idea is that this plugin contains and serves a bundle of the React app, which uses the overview() decorator of the fastify-overview plugin to grab the data and expose it in a route (registered by this plugin) which the UI will use to display the data

### CODE REVIEW

File: src/store/index.jsx [hunk 1]
```diff
@@ -1,23 +1,39 @@
...
-    value: data,
+    value: null,
     isLoading: false,
-    errors: null,
-  });
+    errors: null
+  })
+
+  const fetchOverview = async () => {
+    const data = await fetch(`${window.location.origin}/json-overview`).then(response => response.json()).catch(error => console.log(error))
```
Comment (simoneb) [inline src/store/index.jsx:14]: avoid mixing async await with promise native methods

**Comment by Eomm** [comment]
May I ask why this PR has been closed?

**Review by Eomm — APPROVED** [review]
LGTM!
Thank you very much to push it over the line!

(not sure if it is a regression but the "open vscode" feature does not work locally for me, but this PR is a MUST compared to the it)

**Comment by simoneb** [comment]
> (not sure if it is a regression but the "open vscode" feature does not work locally for me, but this PR is a MUST compared to the it)

I don't see any information about the source code being returned by the fastify-overview plugin, I guess that's the source of the issue

File: example/app.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+'use strict'
+
+const path = require('path')
+const AutoLoad = require('@fastify/autoload')
+
+module.exports = async function (fastify, opts) {
+  await fastify.register(require('fastify-overview'))
```
Comment (Eomm) [inline example/app.js:7]:
```suggestion
  await fastify.register(require('fastify-overview'), {
    addSource: true
  })
```

**Review by Eomm — APPROVED** [review]

**Review by simoneb — APPROVED** [review]

**Pull request merged**